Apps/W1/EDocumentsConnector/app/src/SignUp/APIRequests.Codeunit.al

```diff
@@ -0,0 +1,224 @@
+        exit(SendRequest(HttpRequestMessage, HttpResponseMessage));
+    end;
+
+    // https://<BASE URL>/api/Peppol/inbox?peppolInstanceId=
+    procedure PatchReceivedDocument(EDocument: Record "E-Document"; var HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage): Boolean
+    var
+        SignUpConnectionSetup: Record SignUpConnectionSetup;
+        UriTemplateLbl: Label '%1/api/Peppol/inbox?peppolInstanceId=%2', Comment = '%1 = Service Url, %2 = Peppol Gateway Instance', Locked = true;
+    begin
+        InitRequest(SignUpConnectionSetup, HttpRequestMessage, HttpResponseMessage);
+        HttpRequestMessage := PrepareRequestMsg("Http Request Type"::PATCH, StrSubstNo(UriTemplateLbl, SignUpConnectionSetup.ServiceURL, EDocument."Document Id"));
+        exit(SendRequest(HttpRequestMessage, HttpResponseMessage));
+    end;
+
+    procedure GetMarketPlaceCredentials(var HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage): Boolean
+    var
+        SignUpConnectionSetup: Record SignUpConnectionSetup;
+        SignUpAuth: Codeunit SignUpAuth;
+        BaseUrlTxt: Label '%1/api/Registration/init?EntraTenantId=%2', Locked = true;
+    begin
+        InitRequest(SignUpConnectionSetup, HttpRequestMessage, HttpResponseMessage);
+        HttpRequestMessage := PrepareRequestMsg("Http Request Type"::POST, StrSubstNo(BaseUrlTxt, SignUpAuth.GetRootUrl(), SignUpAuth.GetBCInstanceIdentifier()));
+        exit(SendRequest(HttpRequestMessage, HttpResponseMessage, true));
+    end;
+
+    local procedure InitRequest(var SignUpConnectionSetup: Record SignUpConnectionSetup; var HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage)
+    begin
+        Clear(HttpRequestMessage);
+        Clear(HttpResponseMessage);
+        if not SignUpConnectionSetup.Get() then
+            Error(MissingSetupErr);
+    end;
+
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage): Boolean
+    begin
+        SendRequest(HttpRequestMessage, HttpResponseMessage, false);
+    end;
+
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage; RootRequest: Boolean): Boolean
+    var
+        SignUpAuth: Codeunit SignUpAuth;
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+    begin
+        HttpRequestMessage.GetHeaders(HttpHeaders);
+        if RootRequest then
+            HttpHeaders.Add('Authorization', SignUpAuth.GetRootBearerAuthText())
+        else
+            HttpHeaders.Add('Authorization', SignUpAuth.GetBearerAuthText());
+        exit(HttpClient.Send(HttpRequestMessage, HttpResponseMessage));
+    end;
+
+    local procedure PrepareRequestMsg(pHttpRequestType: Enum "Http Request Type"; Uri: Text) RequestMessage: HttpRequestMessage
```

Comment:
parameter names should be capital. HttpRequestType instead.

Reply:
Done